Hearthstone Deck Tracker/Controls/Overlay/BobsBuddyPanel.xaml.cs

```diff
@@ -170,42 +227,134 @@ public Visibility InfoVisibility
 			}
 		}
 
+		private Visibility _averageDamageInfoVisibility = Visibility.Collapsed;
+		public Visibility AverageDamageInfoVisibility
+		{
+			get => _averageDamageInfoVisibility;
+			set
+			{
+				_averageDamageInfoVisibility = value;
+				OnPropertyChanged();
+			}
+		}
+
+		private Visibility _closeAverageDamageInfoVisibility = Config.Instance.BobsBuddyAverageDamageInfoClosed ? Visibility.Collapsed : Visibility.Visible;
+		public Visibility CloseAverageDamageInfoVisibility
+		{
+			get => _closeAverageDamageInfoVisibility;
+			set
+			{
+				_closeAverageDamageInfoVisibility = value;
+				OnPropertyChanged();
+			}
+		}
 
 		public event PropertyChangedEventHandler PropertyChanged;
 
-		internal void ShowCompletedSimulation(double winRate, double tieRate, double lossRate, double playerLethal, double opponentLethal)
+		private bool _resultsPanelExpanded = false;
+
+		private static List<int> _playerDamageDealtBounds;
+		private static List<int> _opponentDamageDealtBounds;
+
+		internal void ShowCompletedSimulation(double winRate, double tieRate, double lossRate, double playerLethal, double opponentLethal, List<int> possibleResults)
 		{
 			ShowPercentagesHideSpinners();
-
+			_lastCombatPossibilities = possibleResults;
+			SetAverageDamage(possibleResults);
 			WinRateDisplay = string.Format("{0:0.#%}", winRate);
 			TieRateDisplay = string.Format("{0:0.#%}", tieRate);
 			LossRateDisplay = string.Format("{0:0.#%}", lossRate);
 			PlayerLethalDisplay = string.Format("{0:0.#%}", playerLethal);
 			OpponentLethalDisplay = string.Format("{0:0.#%}", opponentLethal);
 
-			PlayerLethalOpacity = playerLethal > 0 ? 1 : 0.3;
-			OpponentLethalOpacity = opponentLethal > 0 ? 1 : 0.3;
+			PlayerLethalOpacity = playerLethal > 0 ? 1 : SoftLabelOpacity;
+			OpponentLethalOpacity = opponentLethal > 0 ? 1 : SoftLabelOpacity;
+
+			PlayerAverageDamageOpacity = possibleResults.Where(x => x > 0).Any() ? 1 : SoftLabelOpacity;
+			OpponentAverageDamageOpacity = possibleResults.Where(x => x < 0).Any() ? 1 : SoftLabelOpacity;
```

Comment:
`Enumerable.Any` accepts a predicate:
```suggestion
			PlayerAverageDamageOpacity = possibleResults.Any(x => x > 0) ? 1 : SoftLabelOpacity;
			OpponentAverageDamageOpacity = possibleResults.Any(x => x < 0) ? 1 : SoftLabelOpacity;
```